fv3core/stencils/d2a2c_vect.py

```diff
@@ -298,84 +296,83 @@ def compute(dord4, uc, vc, u, v, ua, va, utc, vtc):
-        domain_edge_x = (1, grid.njc + 2, grid.npz)
-        jslice = slice(grid.js - 1, grid.je + 2)
-        if grid.west_edge and not grid.nested:
-            vol_conserv_cubic_interp_x(
-                utmp, uc, origin=(i1, j1, 0), domain=domain_edge_x
-            )
-            islice = slice(grid.is_ - 2, grid.is_ + 2)
-            utc[grid.is_, jslice, :] = edge_interpolate4_x(
-                ua[islice, jslice, :], grid.dxa[islice, jslice, :]
-            )
-            uc_x_edge1(
-                utc,
-                grid.sin_sg3,
-                grid.sin_sg1,
-                uc,
-                origin=(i1 + 1, j1, 0),
-                domain=domain_edge_x,
-            )
-            vol_conserv_cubic_interp_x_rev(
-                utmp, uc, origin=(i1 + 2, j1, 0), domain=domain_edge_x
-            )
-            contravariant_stencil(
-                uc,
-                v,
-                grid.cosa_u,
-                grid.rsin_u,
-                utc,
-                origin=(i1, j1, 0),
-                domain=domain_edge_x,
-            )
-            contravariant_stencil(
-                uc,
-                v,
-                grid.cosa_u,
-                grid.rsin_u,
-                utc,
-                origin=(i1 + 2, j1, 0),
-                domain=domain_edge_x,
-            )
-
-        if grid.east_edge and not grid.nested:
-            vol_conserv_cubic_interp_x(
-                utmp, uc, origin=(nx - 1, j1, 0), domain=domain_edge_x
-            )
-            islice = slice(nx - 2, nx + 2)
-            utc[nx, jslice, :] = edge_interpolate4_x(
-                ua[islice, jslice, :], grid.dxa[islice, jslice, :]
-            )
-            uc_x_edge1(
-                utc,
-                grid.sin_sg3,
-                grid.sin_sg1,
-                uc,
-                origin=(nx, j1, 0),
-                domain=domain_edge_x,
-            )
-            vol_conserv_cubic_interp_x_rev(
-                utmp, uc, origin=(nx + 1, j1, 0), domain=domain_edge_x
-            )
-            contravariant_stencil(
-                uc,
-                v,
-                grid.cosa_u,
-                grid.rsin_u,
-                utc,
-                origin=(grid.ie, j1, 0),
-                domain=domain_edge_x,
-            )
-            contravariant_stencil(
-                uc,
-                v,
-                grid.cosa_u,
-                grid.rsin_u,
-                utc,
-                origin=(nx + 1, j1, 0),
-                domain=domain_edge_x,
-            )
+    domain_edge_x = (1, grid.njc + 2, grid.npz)
+    jslice = slice(grid.js - 1, grid.je + 2)
+    if grid.west_edge and not grid.nested:
+        vol_conserv_cubic_interp_x(
+            utmp, uc, origin=(i1, j1, 0), domain=domain_edge_x
+        )
+        islice = slice(grid.is_ - 2, grid.is_ + 2)
+        utc[grid.is_, jslice, -1] = edge_interpolate4_x(
+            ua[islice, jslice, -1], grid.dxa[islice, jslice]
+        )
```

Comment:
oh man I'm so sorry, there is so much python code you have to deal with here that just gets removed by the open PRs! 
I think this may be one of the reasons d2a2c is not validating, the edge interpolate 4 should be called over the whole column, previously was ':'. I think python won't like that though (combining arrays of different shapes), so we might just want to keep it simple and add a loop over k, that'll get blown away anyways when we have the d2a2c as a single stencil

Reply:
Thank you!